kamu-core/src/domain/repos/dataset.rs

```diff
@@ -26,6 +27,9 @@ pub trait Dataset: Send + Sync {
     fn as_checkpoint_repo(&self) -> &dyn ObjectRepository;
     fn as_cache_repo(&self) -> &dyn NamedObjectRepository;
 
+    fn supports_smart_protocol(&self) -> bool;
```

Comment:
I was surprised to see this method here - protocols are an "adapter" concern, so I think we should avoid them leaking into the core domain model.

I was thinking that `http(s)://` will always mean simple transfer and `odf+http(s)://` will indicate smart transfer - so based on the URL sheme we pick the desired protocol and there will be no need to ask dataset what protocol it supports.

Reply:
Resolved